src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -2675,3 +2736,672 @@ def costing_upper_bound_eq(c, p):
 
         # method has finished building components
         b.components_already_built = True
+
+    def calculate_NPV(
+        # flowsheet block or costing block
+        b,
+        # required NPV arguments
+        discount_percentage,
```

Comment:
This is a lot of arguments the user needs to provide to the method. This makes me think it would be better to define these in a `ConfigDict` on the class which the user can then set and play with more easily (and can be better documented), and this method can then access the values from the `ConfigDict`.

Reply:
Thank you for the suggestion. I will look up the proper syntax for implementing this approach.

Reply:
Would the new `CONFIG` variable belong to the `QGESSCostingData` class and include every argument, so I'd reformulate the other methods as well? I'm looking at code examples of implementations and deciding what I need to add to this file.

Reply:
No. You can overload the config from the parent class so only this class sees the changes.

That is in part why most model classes start with `CONFIG = CONFIG()` - note the `()`. By calling the existing `CONFIG` you are actually making a copy of it for the new class so that any changes you make do not affect the parent (but any changes in the parent will propagate down to the child).